guides/keras_hub/getting_started.py

```diff
@@ -1,633 +1,788 @@
-TensorFlow or PyTorch backends with zero changes, simply update the
-`KERAS_BACKEND` below.
+![](https://storage.googleapis.com/keras-nlp/getting_started_guide/prof_keras_evolution.png)
 
-This guide demonstrates our modular approach using a sentiment analysis example at six
-levels of complexity:
-
-* Inference with a pretrained classifier
-* Fine tuning a pretrained backbone
-* Fine tuning with user-controlled preprocessing
-* Fine tuning a custom model
-* Pretraining a backbone model
-* Build and train your own transformer from scratch
+As always, we'll keep our Keras guides focused on real-world code examples. You can play
+with the code here at any time by clicking the Colab link at the top of the guide.
+"""
 
-Throughout our guide, we use Professor Keras, the official Keras mascot, as a visual
-reference for the complexity of the material:
+"""
+## Installation and Setup
+"""
 
-<img src="https://storage.googleapis.com/keras-hub/getting_started_guide/prof_keras_evolution.png" alt="drawing" height="250"/>
+"""
+To begin, let's install keras-hub. The library is available on PyPI, so we can simply
+install it with pip.
 """
 
 """shell
-pip install -q --upgrade keras-hub
-pip install -q --upgrade keras  # Upgrade to Keras 3.
+pip install --upgrade --quiet keras-hub-nightly keras-nightly
+"""
+
+"""
+Keras 3 was built to work on top of TensorFlow, Jax, and Torch backends. You should
+specify the backend first thing when writing Keras code, before any library imports. We
+will use the Jax backend for this guide, but you can use `torch` or `tensorflow` without
+changing a single line in the rest of this guide. That's the power of Keras 3!
+
+We will also set `XLA_PYTHON_CLIENT_MEM_FRACTION`, which frees up the whole GPU for
+Jax to use from the start.
 """
 
 import os
 
 os.environ["KERAS_BACKEND"] = "jax"  # or "tensorflow" or "torch"
+os.environ["XLA_PYTHON_CLIENT_MEM_FRACTION"] = "1.0"
 
-import keras_hub
-import keras
+"""
+Lastly, we need to do some extra setup to access the models used in this guide. Many
+popular open LLMs, such as Gemma from Google and Llama from Meta, require accepting
+a community license before accessing the model weights. We will be using Gemma in this
+guide, so we can follow the following steps:
 
-# Use mixed precision to speed up all training in this guide.
-keras.mixed_precision.set_global_policy("mixed_float16")
+1. Go to the [Gemma 2](https://www.kaggle.com/models/keras/gemma2) model page, and accept
+   the license at the banner at the top.
+2. Generate an Kaggle API key by going to [Kaggle settings](https://www.kaggle.com/settings)
+   and clicking "Create New Token" button under the "API" section.
+3. Inside your colab notebook, click on the key icon on the left hand toolbar. Add two
+   secrets: `KAGGLE_USERNAME` with your username, and `KAGGLE_KEY` with the API key you just
+   created. Make these secrets visible to the notebook you are running.
```

Comment:
NIT: Provide access to these secrets in your notebook when prompted while running the Colab.